examples/ensemble/plot_gradient_boosting_quantile.py

```diff
@@ -3,77 +3,160 @@
+# interval (from 5th to 95th conditional percentiles).
+import matplotlib.pyplot as plt
 
-clf.set_params(loss='ls')
-clf.fit(X, y)
 
-# Make the prediction on the meshed x-axis
-y_pred = clf.predict(xx)
+y_pred = all_models['ls'].predict(xx)
+y_lower = all_models['q 0.05'].predict(xx)
+y_upper = all_models['q 0.95'].predict(xx)
+y_med = all_models['q 0.50'].predict(xx)
 
-# Plot the function, the prediction and the 95% confidence interval based on
-# the MSE
 fig = plt.figure()
 plt.plot(xx, f(xx), 'g:', label=r'$f(x) = x\,\sin(x)$')
-plt.plot(X, y, 'b.', markersize=10, label=u'Observations')
-plt.plot(xx, y_pred, 'r-', label=u'Prediction')
+plt.plot(X_test, y_test, 'b.', markersize=10, label='Test observations')
+plt.plot(xx, y_med, 'r-', label='Predicted median', color="orange")
+plt.plot(xx, y_pred, 'r-', label='Predicted mean')
 plt.plot(xx, y_upper, 'k-')
 plt.plot(xx, y_lower, 'k-')
 plt.fill(np.concatenate([xx, xx[::-1]]),
          np.concatenate([y_upper, y_lower[::-1]]),
-         alpha=.5, fc='b', ec='None', label='95% prediction interval')
+         alpha=.5, fc='b', ec='None', label='Predicted 90% interval')
 plt.xlabel('$x$')
 plt.ylabel('$f(x)$')
 plt.ylim(-10, 20)
 plt.legend(loc='upper left')
 plt.show()
+
+# %%
+# Note that the predicted median is on average below the predicted mean as the
+# noise is skewed towards high values (large outliers). Also note that the
+# median estimate is smoother because of its natural robustness to outliers.
+#
+# Also observe that the inductive bias of gradient boosting trees is
+# unfortunately preventing our 0.05 quantile to fully capture the sinoisoidal
+# shape of the signal, in particular around x=8. Tuning hyper-parameters could
+# potentially reduce this effect a bit.
+#
+# Measure the models with :func:`mean_square_error` and :func:`pinball_loss`
+# metrics on the training dataset.
+from pandas import DataFrame
+
+results = []
+for name, gbr in sorted(all_models.items()):
+    metrics = {'model': name}
+    y_pred = gbr.predict(X_train)
+    for alpha in [0.05, 0.5, 0.95]:
+        metrics["pbl=%1.2f" % alpha] = pinball_loss(
+            y_train, y_pred, alpha=alpha)
+    metrics['MSE'] = mean_squared_error(y_train, y_pred)
+    results.append(metrics)
+DataFrame(results).set_index('model')
+
+# %%
+# One column shows all models evaluated by the same metric. The minimum number
+# on a column should be obtained when the model is trained and measured with
+# the same metric. This should be always the case on the training set if the
+# training converged.
+#
+# Note that because the target noise is skewed by the presence of large
+# outliers, the expected conditional mean and conditional median are
+# signficiantly different and therefore one could not use the least squares
+# model get a good estimation of the conditional median nor the converse.
+#
+# If the target distribution had not been skewed and had no ouliers (e.g. with
```

Comment:
```suggestion
# If the target distribution were symmetric and had no ouliers (e.g. with
```
Just a suggestion. The point is to use the word "symmetric distribution" at least once as this is, in my point of view, the crucial property.
Note also that for large samples, mean and median would always give almost same results for symmetric distributions - even for heavy-tailed once.